design-toolkit/components/src/components/dialog/index.tsx

```diff
@@ -87,6 +87,30 @@ export interface DialogProps extends DialogTriggerProps {
   parentRef?: RefObject<HTMLElement | null>;
 }
 
+/**
+ * Dialog - A modal dialog component for important content and interactions
+ *
+ * Provides accessible modal functionality with focus management, backdrop handling,
+ * and keyboard navigation. Supports multiple sizes and customizable dismissal behavior.
+ * Perfect for confirmations, forms, or any content requiring user focus.
+ *
+ * @example
+ * // Basic dialog with trigger
+ * <Dialog>
+ *   <Button>Open Dialog</Button>
+ *   <Dialog.Body>
+ *     {({ close }) => (
+ *       <>
+ *         <Dialog.Title>Confirm Action</Dialog.Title>
+ *         <p>Are you sure you want to continue?</p>
+ *         <Dialog.Footer>
+ *           <Dialog.Button onPress={close}>Confirm</Dialog.Button>
```

Comment:
Future TODO, get rid of this component and just use Button (and dialog will implement Button.Provider)